app/lib/theme/CustomDiffFields.tsx

```diff
@@ -20,193 +23,161 @@ const contentSuffixElement = (
+};
 
-const overwriteNotification = (
-  specs: OverwriteNotificationArgument
-): string | number | JSX.Element => {
-  const {
-    latestCommittedData,
-    isNumber,
-    isDate,
-    isMoney,
-    isPercentage,
-    numberOfDecimalPlaces,
-  } = specs;
+const renderTooltip = () => {
+  // TODO: implement this
+  return <></>;
+};
 
-  const displayString = "Latest committed value: ";
+const renderArrow = () => {
+  return (
+    <FontAwesomeIcon
+      className={"diff-arrow"}
+      size="lg"
+      color="black"
+      icon={faLongArrowAltRight}
+    />
+  );
+};
 
-  if (!latestCommittedData) {
-    return `${displayString}none`;
+const showStringDiff = (
+  id: string,
+  oldData: string | undefined,
+  newData: string | undefined,
+  latestCommittedData: string | undefined,
+  isDate: boolean,
+  contentSuffix?: string
+): JSX.Element => {
+  const [diffOldClsName, diffNewClsName, diffTextClsName] = [
+    "diffOld",
+    "diffNew",
+    "diffText",
+  ];
+
+  // Case 7 ->  111
+  if (oldData && newData && latestCommittedData) {
+    return (
+      <>
+        {latestCommittedData !== oldData && (
+          <>
+            <span
+              id={id && `${id}-${diffOldClsName}1`}
+              className={diffOldClsName}
+            >
+              {renderData(isDate, latestCommittedData)}
+            </span>
+            {contentSuffix && contentSuffixElement(id, contentSuffix)}
+            {renderArrow()}
+          </>
+        )}
+        <span id={id && `${id}-${diffOldClsName}`} className={diffOldClsName}>
+          {renderData(isDate, oldData)}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+        {renderArrow()}
+        <span id={id && `${id}-${diffNewClsName}`} className={diffNewClsName}>
+          {renderData(isDate, newData)}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+        {renderTooltip()}
+      </>
+    );
   }
-  if (latestCommittedData.isArray && latestCommittedData.length() === 0) {
-    return `${displayString}none`;
+  // case 6 ->  110
+  if (oldData && newData && !latestCommittedData) {
+    return (
+      <>
+        <span id={id && `${id}-${diffOldClsName}`} className={diffOldClsName}>
+          {renderData(isDate, oldData)}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+        {renderArrow()}
+        <span id={id && `${id}-${diffNewClsName}`} className={diffNewClsName}>
+          {renderData(isDate, newData)}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+        {renderTooltip()}
+      </>
+    );
   }
-  if (isDate) {
-    return `${displayString}${getLocaleFormattedDate(latestCommittedData)}`;
+
+  // Case 5 ->  101
+  if (oldData && !newData && latestCommittedData) {
+    return (
+      <>
+        <span id={id && `${id}-${diffOldClsName}`} className={diffOldClsName}>
+          {isDate ? getLocaleFormattedDate(oldData) : oldData}
```

Comment:
There are a few of these lines, can they be replaced with the `renderData` function?